packages/core/src/crypto/JwsService.ts

```diff
@@ -140,22 +149,58 @@ export class JwsService {
     }
 
     return {
+      ...options,
       alg: options.alg,
-      jwk: options.jwk,
+      jwk: options.jwk?.toJson(),
       kid: options.kid,
     }
   }
+
+  private async jwkFromProtectedHeader(agentContext: AgentContext, protectedHeader: JwsProtectedHeader): Promise<Jwk> {
+    if (protectedHeader.jwk && protectedHeader.kid) {
+      throw new AriesFrameworkError(
+        'Both JWK and kid are defined in the protected header. Only one of the two is allowed.'
+      )
+    }
+
+    // Jwk
+    if (protectedHeader.jwk) {
+      return getJwkFromJson(protectedHeader.jwk)
+    }
+
+    // Kid
+    if (protectedHeader.kid) {
+      if (!protectedHeader.kid.startsWith('did:')) {
```

Comment:
Might be good to have a util in AFJ, if it does not already exist, like `isDid(potentialDid: string): bool`.